packages/bpk-component-infinite-scroll/src/DataSource.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+class DataSource {
+  listeners: Array<() => void>;
+
+  onDataChange: (callback: (updatedElements: Array<any>) => void) => boolean;
+
+  offDataChange: (callback: (updatedElements: Array<any>) => void) => boolean;
```

Comment:
Why are these two members listed twice? Both here and below with their implementations.

Reply:
removed this two